### PR TITLE
add value to specify apiversion for rbac settings and set value to v1

### PR DESCRIPTION
**Pull request by transient1**
**Why This PR?**
Kubernetes `1.22` removes `apiVersion` `rbac.authorization.k8s.io/v1beta1 `. This is a change for [incubator/oom-event-generator]

Fixes #

**Changes**
Changes proposed in this pull request:

*set to v1 in rbac rolebinding and clusterrole
*

**Checklist:**

* [x] I have included the name of the chart in the title of this PR in square brackets i.e. `[stable/goldilocks]`.
* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

### CODE REVIEW

**Review by sudermanjr — CHANGES_REQUESTED** [review]
I think we should just set this to `v1`. It's been available since 1.17 I think

File: incubator/oom-event-generator/templates/clusterrole.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/{{ .Values.rbac.apiVersion }}
```
Comment (rbren) [inline incubator/oom-event-generator/templates/clusterrole.yaml]: I would suggest something like this instead (there are a few other examples of this around the charts repo)
```suggestion
{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1" }}
apiVersion: rbac.authorization.k8s.io/v1
{{- else }}
apiVersion: rbac.authorization.k8s.io/v1beta1
{{- end }}
```

Comment (transient1) [reply incubator/oom-event-generator/templates/clusterrole.yaml]: I hard-coded it because Andy got to me first!

**Review by sudermanjr — APPROVED** [review]

**Comment by rbren** [comment]
Makes sense to me! We should probably change for all the incubator charts - there's a few more it looks like

**Comment by rbren** [comment]
Dunno why helm-docs is angry, but if you run `helm-docs --sort-values-order=file` it should fix it

**Pull request merged**